### PR TITLE
Update Java version

### PR DESCRIPTION
**Pull request by gcnyin**
**Summary of changes**
Changes introduced in this pull request:
- 

Update Java version.

- hotspot to Java 17

**Reference issue to close (if applicable)**

**Other information and links**

### CODE REVIEW

**Comment by LesnyRumcajs** [comment]
Thanks, sounds good. How about bumping the image in Java benches as well?

**Comment by gcnyin** [comment]
> Thanks, sounds good. How about bumping the image in Java benches as well?

Good idea. Let me do it.

**Comment by gcnyin** [comment]
@LesnyRumcajs Could you please review again?

File: java_hotspot_grpc_g1gc_bench/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM adoptopenjdk:16_36-jdk-hotspot
+FROM openjdk:17
```
Comment (LesnyRumcajs) [inline java_hotspot_grpc_g1gc_bench/Dockerfile:1]: Does the name "hotspot" still make sense here for those java benchmarks? Or should we remove this from their names?

Comment (LesnyRumcajs) [inline java_hotspot_grpc_g1gc_bench/Dockerfile:1]: Does the name "hotspot" still make sense here for those java benchmarks? Or should we remove this from their names?

**Comment by gcnyin** [comment]
@LesnyRumcajs Actually, I don't think it's necessary to add OpenJ9 jdk test cases. No one(maybe abount 99.999%) is actually using it in production. Just keeping openjdk(hotspot) is enough.

**Comment by LesnyRumcajs** [comment]
> @LesnyRumcajs Actually, I don't think it's necessary to add OpenJ9 jdk test cases. No one(maybe about 99.999%) is actually using it in production. Just keeping openjdk(hotspot) is enough.

Thanks for the explanation, I wasn't sure if hotspot is the same thing in this context.

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**